### PR TITLE
Aways callback implementation

### PR DESCRIPTION
**Pull request by heat**
implementation proposal for #1

### CODE REVIEW

File: src/main/java/org/turbogwt/net/http/FluentRequest.java [hunk 1]
```diff
@@ -141,6 +141,16 @@
...
 
     /**
+     * Define a callback to be called when request is either resolved or rejected.
+     * <br />
+     * The argument is a Callback. When the request is resolved or rejected, the callback is executed. <br />
+     * When the request is resolved or rejected, the callback is executed using the request and response arguments.
+     * If it was failured request the response parameter is null.
+     * @param callback
+     */
+    FluentRequestSender<RequestType, ResponseType> aways(SingleCallback callback);
```
Comment (reinert) [inline src/main/java/org/turbogwt/net/http/FluentRequest.java]: Typo! _aways_ should be _always_

File: src/main/java/org/turbogwt/net/http/FluentRequest.java [hunk 2]
```diff
@@ -141,6 +141,16 @@
     FluentRequestSender<RequestType, ResponseType> on(int statusCode, SingleCallback callback);
 
     /**
+     * Define a callback to be called when request is either resolved or rejected.
+     * <br />
+     * The argument is a Callback. When the request is resolved or rejected, the callback is executed. <br />
```
Comment (reinert) [inline src/main/java/org/turbogwt/net/http/FluentRequest.java:146]: Please do not repeat words unnecessarily. Docs should be as succinct as possible.

File: src/main/java/org/turbogwt/net/http/FluentRequestImpl.java [hunk 3]
```diff
@@ -26,8 +26,7 @@
 import com.google.gwt.http.client.Response;
 import com.google.gwt.user.client.rpc.AsyncCallback;
 
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
```
Comment (reinert) [inline src/main/java/org/turbogwt/net/http/FluentRequestImpl.java]: Please, explicit imports.

**Comment by reinert** [comment]
Your PR failed due to checktyle errors. Please fix them and commit as "Checkstyle issues",

File: src/main/java/org/turbogwt/net/http/FluentRequestImpl.java [hunk 4]
```diff
@@ -64,6 +65,7 @@
     private UriBuilder uriBuilder;
     private String uri;
     private JsMap<SingleCallback> mappedCallbacks;
+    private List<SingleCallback> alwaysCallbacks;
```
Comment (reinert) [inline src/main/java/org/turbogwt/net/http/FluentRequestImpl.java]: I prefer to use JsArray for performance reasons (although minimal).

File: src/test/java/org/turbogwt/net/http/FluentRequestImplTest.java [hunk 5]
```diff
@@ -417,4 +462,20 @@ public void testFormDataRequest() {
         final RequestMock requestMock = ServerStub.getRequestData(uri);
         assertEquals(serialized, requestMock.getData());
     }
+
+    class MutableBoolean {
```
Comment (reinert) [inline src/test/java/org/turbogwt/net/http/FluentRequestImplTest.java:463]: Please use boolean array. If you find better to use this MultableBoolean, then create a new PR refactoring all boolean arrays to it.

**Comment by heat** [comment]
ok, all right in Travis.

**Comment by reinert** [comment]
Very nice!

**Pull request merged**